projects/core/src/anonymous-consents/services/anonymous-consents-state-persistance.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+  toggleTemplatesUpdated = jasmine.createSpy();
+}
+
+const mockConsent: AnonymousConsent = {
+  templateCode: 'consentCode',
+  templateVersion: 1,
+  consentState: ANONYMOUS_CONSENT_STATUS.GIVEN,
+};
+
+const mockState: SyncedAnonymousConsentsState = {
+  templates: {
+    value: [
+      {
+        id: 'consentId',
+        name: 'consentName',
+      },
+    ],
+  },
+  consents: [mockConsent],
+  ui: {
+    bannerDismissed: false,
+    updated: false,
+  },
+};
+
+describe('AnonymousConsentsStatePersistenceService', () => {
+  let service: AnonymousConsentsStatePersistenceService;
+  let persistenceService: StatePersistenceService;
+  let store: Store<StateWithAnonymousConsents>;
+  let anonymousConsentsService: AnonymousConsentsService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        StoreModule.forRoot({}),
+        StoreModule.forFeature(
+          ANONYMOUS_CONSENTS_STORE_FEATURE,
+          fromAnonymousConsentsReducers.getReducers()
+        ),
+      ],
+      providers: [
+        AnonymousConsentsStatePersistenceService,
+        StatePersistenceService,
+        {
+          provide: AnonymousConsentsService,
+          useClass: MockAnonymousConsentsService,
+        },
+      ],
+    });
+
+    service = TestBed.inject(AnonymousConsentsStatePersistenceService);
+    persistenceService = TestBed.inject(StatePersistenceService);
+    store = TestBed.inject(Store);
+    anonymousConsentsService = TestBed.inject(AnonymousConsentsService);
+
+    spyOn(store, 'dispatch').and.stub();
+    spyOn(persistenceService, 'syncWithStorage').and.stub();
+  });
+
+  it('should inject service', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('initSync()', () => {
+    it('should sync state with storage', () => {
+      service.initSync();
+
+      expect(persistenceService.syncWithStorage).toHaveBeenCalled();
+    });
+  });
+
+  describe('getAuthState()', () => {
+    it('should return the full state', (done: DoneFn) => {
+      spyOn(store, 'select').and.returnValue(of(mockState));
+
+      service['getAuthState']().subscribe((state) => {
+        expect(state).toEqual(mockState);
+
+        done();
+      });
+    });
+
+    it('should return the state partially', (done: DoneFn) => {
+      const partialState = { ...mockState };
+      delete partialState.ui;
+
+      spyOn(store, 'select').and.returnValue(of(partialState));
+
+      service['getAuthState']().subscribe((state) => {
+        expect(state).toEqual(partialState);
+
+        done();
+      });
+    });
+  });
+
+  describe('onRead()', () => {
+    it('should handle templates', () => {
+      service['onRead']({ templates: mockState.templates });
+
+      expect(anonymousConsentsService.getTemplates).toHaveBeenCalled();
+    });
+
+    it('should handle consents', () => {
+      service['onRead']({ consents: mockState.consents });
+
+      expect(anonymousConsentsService.setConsents).toHaveBeenCalled();
```

Comment:
Can we check in those tests if we passed correct payload to the methods as well?